packages/vite/src/node/plugins/css.ts

```diff
@@ -307,6 +320,25 @@ export function cssPlugin(config: ResolvedConfig): Plugin {
         map,
       }
     },
+
+    async renderChunk(_, chunk) {
+      if (!rootVarsEmitted && rootVarsCache.size) {
+        const rootVarsCss = joinRootVars(rootVarsCache)
+        chunk.viteMetadata!.importedCss.add(
+          this.getFileName(
+            this.emitFile({
+              name: 'root-css-variable.css',
```

Comment:
I think it better to inject the css in the css chunk, because this css will biger and biger. and the user can't do the loading on demand

Reply:
As a supplement, the `index.html` will load 100 assets, but it only need the 10 assets

Reply:
When I think further,  it would be better to keep the transform behavior both in dev and build mode. Otherwise, it may break some plugins that depend on base64 during the `transform` phase. 

Duplicate base64 CSS variables in a CSS chunk will be removed in `css-post` plugin's `renderChunk` phase.